pkg/apply/resource_infos.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package apply
+
+import (
+	"sort"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/resource"
+)
+
+type ResourceInfos []*resource.Info
+
+var _ sort.Interface = ResourceInfos{}
+
+func (a ResourceInfos) Len() int      { return len(a) }
+func (a ResourceInfos) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a ResourceInfos) Less(i, j int) bool {
+	x := a[i].Object.GetObjectKind().GroupVersionKind()
+	o := a[j].Object.GetObjectKind().GroupVersionKind()
+	if !Equals(x, o) {
+		return IsLessThan(x, o)
+	}
+	return a[i].String() < a[j].String()
```

Comment:
So `Info` contains a lot of fields, and it is not obvious what we are doing when comparing the `String` functions. Also, we end up depending on the implementation of `String`, which could easily change. I suggest we should be more explicit about what we are doing here.

Reply:
This is just a tie-breaker and if this line is being executed, the result shouldn't matter as both options are same and their ordering doesn't hurt. This is how it is done in kustomize currently and I borrowed the logic to be consistent.